[Flight Recorder] Reverted to include stack traces for dump pipe triggered FR dump

VieEeEw opened the pull request:
Summary: We should also retry if include stacktraces failed.

Test Plan: eyes

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci

fduwjj reviewed (changes requested):
why is that? This is intentionally make it false. Re-try is useless

fduwjj reviewed (approved):
ok previous change does break this, we can revert it back.

VieEeEw committed:
[Flight Recorder] Reverted to include stack traces for dump pipe triggered FR dump (#167023)

Summary:

We should also retry if include stacktraces failed. Changed was introduced in https://github.com/pytorch/pytorch/pull/164591

Test Plan: eyes

Reviewed By: fduwjj

Differential Revision: D86248484

jeanschmidt commented:
Differential Revision: D86248484

wang-mask commented:
Thanks for the discussion.


My concern is less about whether stacks are nice to have, and more about **dump reliability when the process is already unhealthy**.

Flight Recorder dumps are critical for diagnosing NCCL hang / watchdog timeouts: without per-rank FR traces, it is often impossible to tell whether the root cause is CPU divergence, collective mismatch, GPU hang, or a network issue.

## The core problem

FR dump with stacks goes through `torch::symbolize()`, which may need the GIL.
If the training process cannot acquire the GIL (or is stuck holding it), a with-stack dump can hang / fail to complete.

This risk is already recognized in-tree:
- `dumpExtraDebuggingInfo()` intentionally calls `dumpDebuggingInfo(false, …)` “to prevent deadlock due to GIL”
- the timeout dump path retries with `includeStackTrace=false` after waiting

But the named-pipe path currently hardcodes:
```cpp
dumpDebuggingInfo(true, onlyActive);
```

@VieEeEw @fduwjj
